openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/vitals/PatientDashboardVitalsPresenter.java

```diff
@@ -23,22 +23,38 @@
 import org.openmrs.mobile.models.Encounter;
 import org.openmrs.mobile.utilities.NetworkUtils;
 
+import rx.android.schedulers.AndroidSchedulers;
+
 public class PatientDashboardVitalsPresenter extends PatientDashboardMainPresenterImpl implements PatientDashboardContract.PatientVitalsPresenter {
 
     private EncounterDAO encounterDAO;
+    private PatientDAO patientDao;
+    private String patientId;
     private PatientDashboardContract.ViewPatientVitals mPatientVitalsView;
 
     public PatientDashboardVitalsPresenter(String id, PatientDashboardContract.ViewPatientVitals mPatientVitalsView) {
-        this.mPatient = new PatientDAO().findPatientByID(id);
+        this.patientId = id;
         this.mPatientVitalsView = mPatientVitalsView;
         this.mPatientVitalsView.setPresenter(this);
         this.encounterDAO = new EncounterDAO();
+        this.patientDao = new PatientDAO();
     }
 
     @Override
     public void start() {
-        loadVitalsFromDB();
-        loadVitalsFromServer();
+        if (this.mPatient == null) {
```

Comment:
we definitely don't want to do that... if there be patient it's data won't be updated